var/spack/repos/builtin/packages/dealii/package.py

```diff
@@ -68,13 +68,20 @@ class Dealii(CMakePackage):
             description='Compile with Python bindings')
     variant('int64',    default=False,
             description='Compile with 64 bit indices support')
+    variant('optflags', default=False,
+            description='Compile using additional optimization flags')
 
     # required dependencies, light version
     depends_on("blas")
     # Boost 1.58 is blacklisted, see
     # https://github.com/dealii/dealii/issues/1591
     # Require at least 1.59
     # +python won't affect @:8.4.2
+    # FIXME: once conretizer can unite unconditional and
```

Comment:
I have a feeling that the issue is the same as with
```
depends_on('foo')
depends_on('foo+mpi', when='+mpi')
```
Namely, once concretizer settles on unconditional dependencies, it can't extend/contract the space of variants.